### PR TITLE
Restructure development pages

### PR DESCRIPTION
**Pull request by moorscode**
Remove the confusing "standards" entry-point and restructure the underlaying items into categories (that were already softly present in the overview page).

Goes along with: https://github.com/Yoast/developer-docs/pull/81

### CODE REVIEW

File: sidebars.js [hunk 1]
```diff
@@ -120,6 +119,16 @@ module.exports = {
...
+
+				{
+					type: 'category',
+					label: 'WP CLI',
+					items: [
+						'features/wp-cli/reindex-indexables',
+					],
+				},
+
+				'features/installation/using-composer',
```
Comment (jcomack) [inline sidebars.js:131]: Please reconsider these based on the CR done for the docs.

File: sidebars.js [hunk 2]
```diff
@@ -178,21 +187,34 @@ module.exports = {
 		},
 		{
 			type: 'category',
-			label: 'Standards',
+			label: 'Plugin development',
```
Comment (jcomack) [inline sidebars.js:190]: This should be Development if we want to be consistent with the permalink / slug.

File: sidebars.js [hunk 3]
```diff
@@ -152,7 +150,6 @@ module.exports = {
 					items: [
 						'customization/yoast-seo/api-filter-actions-deprecations',
 						'customization/yoast-seo/adding-custom-data-analysis',
-						'customization/yoast-seo/indexables-cli',
```
Comment (jcomack) [inline sidebars.js:146]: Why was this entry completely removed instead of being moved?

Comment (moorscode) [reply sidebars.js:146]: This is moved to the `features/wp-cli/indexables-cli` path.
That resulted in the wrongly named tmp file I think 😊

**Comment by moorscode** [comment]
@jono-alderson could you help out with the redirects for this PR? 
Ive placed them in this sheet: https://docs.google.com/spreadsheets/d/1a2bAhI_P2l3kNgX252txQmHfVLb4fAh2765aUEWKAYo/edit?usp=sharing

**Pull request merged**